src/compiler/ruby_generator.cc

```diff
@@ -37,16 +38,58 @@ using std::vector;
 namespace grpc_ruby_generator {
 namespace {
 
+// returns the full canonical message name
+grpc::string GetCanonicalMessageType(const Descriptor* msg) {
+  // If msg is nested, find the topmost message
+  const Descriptor* top_level_msg = msg;
+  while (top_level_msg->containing_type()) {
```

Comment:
nit: can we make this a `top_levle_msg->containing_type() != nullptr`

can we also illustrate the need for that logic with another test?

Reply:
can we please make this check on line 45 the `nullptr` check, and revert the `nullptr` check on line 51?